src/js/utils/server-safe-globals.ts

```diff
@@ -79,59 +84,18 @@ const globalThisShim = {
       media,
     };
   },
-};
+} as unknown as typeof globalThis;
 
 export const isServer =
   typeof window === 'undefined' || typeof window.customElements === 'undefined';
 
 const isShimmed = Object.keys(globalThisShim).every((key) => key in globalThis);
 
-/**
- * @type { globalThis & {
- *   WebKitPlaybackTargetAvailabilityEvent?,
- *   chrome?,
- *   DocumentFragment?,
- *   getComputedStyle,
- *   CastableVideoElement?
- * } |
- * {Node,
- * HTMLElement,
- * customElements,
- * CustomEvent,
- * getComputedStyle,
- * addEventListener?,
- * removeEventListener?,
- * localStorage?,
- * WebKitPlaybackTargetAvailabilityEvent?,
- * window?,
- * document?,
- * chrome?,
- * DocumentFragment?,
- * ResizeObserver?,
- * CastableVideoElement?,
- * navigator?,
- * matchMedia,
- * } }
- * */
-export const GlobalThis = isServer && !isShimmed ? globalThisShim : globalThis;
-
-/**
- * @type { document & { webkitExitFullscreen? } |
- * {createElement,
- * createElementNS?,
- * activeElement?,
- * fullscreenElement?,
- * webkitExitFullscreen?,
- * getElementById?,
- * pictureInPictureElement?,
- * exitPictureInPicture?,
- * pictureInPictureEnabled?,
- * requestPictureInPicture?,
- * addEventListener?,
- * removeEventListener?,
- * } }
- */
-export const Document =
-  isServer && !isShimmed ? documentShim : globalThis.document;
+export const GlobalThis: typeof globalThis =
```

Comment:
updated these to "lie" on their export types as if they are always running in the browser and not minimal shim impls when on e.g. node. That does mean we won't get type errors for cases where we have failed to shim/polyfill.